### PR TITLE
Autocomplete interface guideline updates

### PR DESCRIPTION
**Pull request by langermank**
Minor updates to Autocomplete. This guideline should be re-addressed after the [accessibility review](https://github.com/github/primer/issues/519) in PRC is completed.

- Update images to use new `focus` 
- Update anatomy to include `leadingVisual` and clear button

Closes https://github.com/github/primer/issues/673

### CODE REVIEW

File: content/components/toggle-switch.mdx [hunk 1]
```diff
@@ -17,81 +17,138 @@ import {Box, Text} from '@primer/components'
...
+    role="presentation"
+    src="https://user-images.githubusercontent.com/2313998/159587115-5ffcdefb-f5e9-4180-8dc3-5328b0440f7f.png"
+  />
 </Box>
 
 <DoDontContainer>
   <Do>
-    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/159587088-99e8fe4c-59a3-4452-836d-bd6f2297045a.png" />
+    <img
+      role="presentation"
```
Comment (owenniblock) [inline content/components/toggle-switch.mdx:47]: I feel like this image adds important context to the message (my providing examples). We should consider adding alt text here.

File: content/components/toggle-switch.mdx [hunk 2]
```diff
@@ -17,81 +17,138 @@ import {Box, Text} from '@primer/components'
...
+      description about the setting that is being turned on or off.
+    </Box>
+    <Box as="p" mt="0">
+      Labels should be succinct, and can describe an object or an action (for example: "Discussions" or "Automatically
+      watch repositories"). Don't use phrases that describe the switch's state in your label.
+    </Box>
+  </div>
+  <img
+    width="456"
+    role="presentation"
```
Comment (owenniblock) [inline content/components/toggle-switch.mdx:39]: I feel like this image adds important context to the message (my providing examples). We should consider adding alt text here.

File: content/components/toggle-switch.mdx [hunk 3]
```diff
@@ -17,81 +17,138 @@ import {Box, Text} from '@primer/components'
...
+    <img
+      role="presentation"
+      src="https://user-images.githubusercontent.com/2313998/159587088-99e8fe4c-59a3-4452-836d-bd6f2297045a.png"
+    />
     <Caption>Use succinct phrases with verbs or nouns</Caption>
   </Do>
   <Dont>
-    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/159587096-5f8a6913-df95-4ea0-9838-a366faf95e52.png" />
+    <img
+      role="presentation"
```
Comment (owenniblock) [inline content/components/toggle-switch.mdx:54]: I feel like this image adds important context to the message (my providing examples). We should consider adding alt text here.

File: content/components/toggle-switch.mdx [hunk 4]
```diff
@@ -114,14 +171,19 @@ If you have a group of configuration options that can be disabled, use a button
 
 ![Configuration options section that can be enabled or disabled. Configuration controls are hidden when it is disabled.](https://user-images.githubusercontent.com/2313998/159587120-2d842830-18c7-4429-bd10-f5424f6fac9e.png)
 
-
 <DoDontContainer>
   <Do>
-    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/159587100-1a1af5b2-dd19-456a-a0ed-c2f9fa7f7c2e.png" />
+    <img
+      role="presentation"
```
Comment (owenniblock) [inline content/components/toggle-switch.mdx:177]: I feel like this image adds important context to the message (my providing examples). We should consider adding alt text here.

File: content/components/toggle-switch.mdx [hunk 5]
```diff
@@ -114,14 +171,19 @@ If you have a group of configuration options that can be disabled, use a button
...
+      src="https://user-images.githubusercontent.com/2313998/159587100-1a1af5b2-dd19-456a-a0ed-c2f9fa7f7c2e.png"
+    />
     <Caption>Use a button to affect multiple related toggles</Caption>
   </Do>
   <Dont>
-    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/159587101-446548c2-41c3-45c1-b710-db3149e2ebe4.png" />
+    <img
+      role="presentation"
```
Comment (owenniblock) [inline content/components/toggle-switch.mdx:184]: I feel like this image adds important context to the message (my providing examples). We should consider adding alt text here. Specifically here we could describe the toggle switch in place of the button.

File: content/components/tokens.mdx [hunk 6]
```diff
@@ -23,99 +36,167 @@ import {LabelGroup, Label, Box} from '@primer/components'
...
+
+<Box
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 2}}
+>
+  <img
+    width="456"
+    alt="token anatomy diagram"
```
Comment (owenniblock) [inline content/components/tokens.mdx:56]: I think we either need to describe the diagram (if we feel the user would benefit from that context on top of the following description of the fields) or we need to make this a presentation image / give it an empty alt text.

File: content/components/tokens.mdx [hunk 7]
```diff
@@ -23,99 +36,167 @@ import {LabelGroup, Label, Box} from '@primer/components'
...
-        <Box as="p">A group of tokens should not have tokens of different sizes.</Box>
+<Box
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="tokens rendered at all sizes"
```
Comment (owenniblock) [inline content/components/tokens.mdx:95]: We should list those sizes here, there is no textual way for users to see the different sizing.

File: content/components/tokens.mdx [hunk 8]
```diff
@@ -23,99 +36,167 @@ import {LabelGroup, Label, Box} from '@primer/components'
...
+
+<Box
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="tokens rendered at all sizes"
```
Comment (owenniblock) [inline content/components/tokens.mdx:120]: This alt text is incorrect, should describe the flow of the tokens.

File: content/components/tokens.mdx [hunk 9]
```diff
@@ -23,99 +36,167 @@ import {LabelGroup, Label, Box} from '@primer/components'
...
```
Comment (owenniblock) [inline content/components/tokens.mdx:143]: This alt text is incorrect, it should describe the image shown.

File: content/components/tokens.mdx [hunk 10]
```diff
@@ -23,99 +36,167 @@ import {LabelGroup, Label, Box} from '@primer/components'
...
-    <Box as="p" mt="0">If the input is being used in a spot where the input height cannot grow, tokens can be forced to a single line that scrolls horizontally.</Box>
+<Box
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="tokens rendered at all sizes"
```
Comment (owenniblock) [inline content/components/tokens.mdx:166]: This alt text is incorrect, it should describe the image shown.

File: content/ui-patterns/feature-onboarding.mdx [hunk 11]
```diff
@@ -105,7 +108,10 @@ Treat feature onboarding as a story or a journey with a beginning, middle, and e
 
 <Flex alignItems="center" flexDirection="column" my={3}>
   <BorderBox width={9 / 12} mb="0" overflow="hidden">
-    <img src="https://user-images.githubusercontent.com/66705495/92781977-7348a100-f359-11ea-9ed1-924f4ea8567a.png" alt="Example of the success state toast component " />
+    <img
+      src="https://user-images.githubusercontent.com/66705495/92781977-7348a100-f359-11ea-9ed1-924f4ea8567a.png"
+      alt="Example of the success state toast component "
```
Comment (owenniblock) [inline content/ui-patterns/feature-onboarding.mdx:113]: We repeat the prefix "Example of" many times in our alt text. We should consider dropping this to reduce verbosity:

```suggestion
      alt="The success state toast component"
```

We should also review our alt text and ensure that any relevant context is added for screen reader users.

File: content/ui-patterns/saving.mdx [hunk 12]
```diff
@@ -91,27 +98,40 @@ When defining these calls to actions, make sure to:
...
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+  <Box mb={3} display="flex" alignItems="center" flexDirection="column" mx="auto" sx={{gap: 2}}>
+    <img
+      width="456"
+      src="https://user-images.githubusercontent.com/2313998/152432687-e119f44c-9325-4ae7-b277-e8559a4eb37b.png"
```
Comment (owenniblock) [inline content/ui-patterns/saving.mdx:112]: Missing alt tag of presentation role, screen reader will attempt to read the URL.

File: content/ui-patterns/saving.mdx [hunk 13]
```diff
@@ -91,27 +98,40 @@ When defining these calls to actions, make sure to:
...
+      alt="A small form in a dialog that has 'Cancel' and 'Save' buttons at the bottom right"
+      src="https://user-images.githubusercontent.com/2313998/151623724-d59d5967-5484-4e81-a4c2-1109409956ee.png"
+    />
+    <Text as="p" mt="0" align="center" fontSize={1}>
+      A form with a button to save and a button to cancel
+    </Text>
+  </Box>
+  <Box mb={3} display="flex" alignItems="center" flexDirection="column" mx="auto" sx={{gap: 2}}>
+    <img
+      width="456"
```
Comment (owenniblock) [inline content/ui-patterns/saving.mdx:128]: Missing alt tag of presentation role, screen reader will attempt to read the URL.

File: content/ui-patterns/saving.mdx [hunk 14]
```diff
@@ -241,27 +304,45 @@ If the change is not obvious within the user's viewport, provide feedback to let
...
+  <Box
+    display="inline-block"
+    overflow="hidden"
+    borderWidth="1px"
+    borderColor="border.default"
+    borderStyle="solid"
+    borderRadius="12px"
+  >
+    <img
+      width="960"
```
Comment (owenniblock) [inline content/ui-patterns/saving.mdx:328]: Missing alt tag of presentation role, screen reader will attempt to read the URL.

**Review by owenniblock — CHANGES_REQUESTED** [review]
I appreciate that all of the accessibility concerns I have here already exist within the documents.

I suggest for most of these issues, we record them and look at resolving them outside of this PR.

However I would like us to resolve the 3 examples I found of this issue if possible:

> Missing alt tag of presentation role, screen reader will attempt to read the URL.

File: content/components/autocomplete.mdx [hunk 15]
```diff
@@ -3,86 +3,143 @@ title: Autocomplete
...
+    as="a"
+    href="https://primer.style/components/Autocomplete"
+    variant="secondary"
+    size="large"
+    style="text-decoration: none;"
+  >
+    <img
+      width="16"
+      height="16"
+      alt=" "
```
[the remark this answers is not in the document]
Comment (langermank) [reply content/components/autocomplete.mdx:23]: I removed the space here: https://github.com/primer/design/pull/267

**Comment by owenniblock** [comment]
@langermank no worries. It would be great if you can reference this PR in the new PR so whoever is on accessibility FR duties can get the context.

I'm happy for accessibility fixes to go in a separate PR - keep things as simple as possible I just wanted to record these issues when I saw them 😄

**Pull request closed without merging**